src/Compatibility/Microsoft.DotNet.PackageValidation/BaselinePackageValidator.cs

```diff
@@ -0,0 +1,126 @@
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    /// <summary>
+    /// Validates that no target framework / rid support is dropped in the latest package.
+    /// Reports all the breaking changes in the latest package.
+    /// </summary>
+    public class BaselinePackageValidator
+    {
+        private static HashSet<string> s_diagList = new HashSet<string>{ DiagnosticIds.TargetFrameworkDropped, DiagnosticIds.TargetFrameworkAndRidPairDropped };
+
+        private readonly Package _baselinePackage;
+        private readonly bool _runApiCompat;
+        private readonly ApiCompatRunner _apiCompatRunner;
+        private readonly ILogger _log;
+        private readonly DiagnosticBag<IDiagnostic> _diagnosticBag;
+
+        public BaselinePackageValidator(Package baselinePackage, string noWarn, (string, string)[] ignoredDifferences, bool runApiCompat, ILogger log)
+        {
+            _baselinePackage = baselinePackage;
+            _runApiCompat = runApiCompat;
+            _log = log;
+            _apiCompatRunner = new(noWarn, ignoredDifferences, _log);
+            _diagnosticBag = new(noWarn?.Split(';')?.Where(t => s_diagList.Contains(t)), ignoredDifferences);
+        }
+
+        /// <summary>
+        /// Validates the latest nuget package doesnot drop any target framework/rid and does not introduce any breaking changes.
+        /// </summary>
+        /// <param name="package">Nuget Package that needs to be validated.</param>
+        public void Validate(Package package)
+        {
+            foreach (ContentItem baselineCompileTimeAsset in _baselinePackage.CompileAssets)
+            {
+                NuGetFramework baselineTargetFramework = (NuGetFramework)baselineCompileTimeAsset.Properties["tfm"];
+                ContentItem latestCompileTimeAsset = package.FindBestCompileAssetForFramework(baselineTargetFramework);
+                if (latestCompileTimeAsset == null)
+                {
+                    if (!_diagnosticBag.Contain(DiagnosticIds.TargetFrameworkDropped, baselineTargetFramework.ToString()))
+                    {
+                        string message = string.Format(Resources.MissingTargetFramework, baselineTargetFramework.ToString());
+                        _log.LogError(DiagnosticIds.TargetFrameworkDropped + " " + message);
+                    }
+                }
+                else if (_runApiCompat)
+                {
+                    _apiCompatRunner.QueueApiCompat(_baselinePackage.PackagePath,
+                        baselineCompileTimeAsset.Path,
+                        package.PackagePath,
+                        latestCompileTimeAsset.Path,
+                        Path.GetFileName(package.PackagePath),
+                        Resources.BaselineVersionValidatorHeader,
+                        string.Format(Resources.MissingApisForFramework, baselineTargetFramework.ToString()));
```

Comment:
`ApiCompat` errors doesn't always mean that there are APIs missing. Should we instead have a more generic error like: `There are API Compatibility errors in between compile asset 'ref/netstandard2.0`(left) and runtime asset `runtime/win-x64/lib/netcoreapp3.1` (right)` ?